mypyc/genops.py

```diff
@@ -1126,23 +1213,153 @@ def generate_attr_defaults(self, cdef: ClassDef) -> None:
         self.functions.append(ir)
         cls.methods[ir.name] = ir
 
+    def load_non_ext_class(self, ir: ClassIR, line: int) -> Value:
+        assert self.non_ext_dict is not None
+        assert self.non_ext_anns is not None
+        assert self.non_ext_bases is not None
+
+        cls_name = self.load_static_unicode(ir.name)
+
+        # Add __annotations__ to the class dict.
+        self.primitive_op(dict_set_item_op,
+                          [self.non_ext_dict, self.load_static_unicode('__annotations__'),
+                           self.non_ext_anns], -1)
+
+        # We add a __doc__ attribute so if the non-extension class is decorated with the
+        # dataclass decorator, dataclass will not try to look for __text_signature__.
+        # https://github.com/python/cpython/blob/3.7/Lib/dataclasses.py#L957
+        filler_doc_str = 'filler docstring for classes decorated with dataclass'
+        self.primitive_op(dict_set_item_op,
+                         [self.non_ext_dict, self.load_static_unicode('__doc__'),
+                          self.load_static_unicode(filler_doc_str)], line)
```

Comment:
If there is a user-defined docstring, will it override it?

Reply:
mypyc currently does not preserve user-defined docstrings (although this should probably happen eventually). Therefore, this is a non-issue right now.